Add files via upload

pierre6677 opened the pull request:
When the searched text is included into a bounded box (a button for example), OCR cannot find the text location. I just added a function removing contours larger than 300. I used  OPENCV contour detection and draw features.

RaiMan commented:
Thanks for now.

I am currently working in the repo SikuliX1 / branch release_2.0.x. So I will check your implementation and most probably add it appropriately modified.

Do you have an example, where it did not work?

pierre6677 commented:
Hello,

I worked on the sikuli robot framework library. I add a few keywords
adaptation for ocr purpose. I will pull the change I made including the
click'in demo as example.

BR.
Pierre

Le jeu. 12 nov. 2020 à 13:28, Raimund Hocke <notifications@github.com> a
écrit :

> Thanks for now.
>
> I am currently working in the repo SikuliX1 / branch release_2.0.x. So I
> will check your implementation and most probably add it appropriately
> modified.
>
> Do you have an example, where it did not work?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RaiMan/SikulixAPI/pull/1#issuecomment-726047703>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG77ADZ2N5J2LMA7PMGJCBDSPPIF7ANCNFSM4TSHHB4Q>
> .
>

pierre6677 commented:
The original click'in demo from sikuli robot framework does not work when
using the "click on text" feature. When the text is inside a boundary box,
the searched text cannot be find. With the modification I made, it works.
It seems other people faced the same issue.

I Also had the problem testing gui automation with robot framework.



Le jeu. 12 nov. 2020 à 13:28, Raimund Hocke <notifications@github.com> a
écrit :

> Thanks for now.
>
> I am currently working in the repo SikuliX1 / branch release_2.0.x. So I
> will check your implementation and most probably add it appropriately
> modified.
>
> Do you have an example, where it did not work?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RaiMan/SikulixAPI/pull/1#issuecomment-726047703>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG77ADZ2N5J2LMA7PMGJCBDSPPIF7ANCNFSM4TSHHB4Q>
> .
>

RaiMan commented:
@pierre6677 
I made some tests with 2.0.4 and the dev version 2.0.5: I had no problems.

So again: please provide sample screenshots and sample images/texts.

pierre6677 commented:
[image: image.png]
Hello,
Sorry for the delay.
I'm working with the same packages.
Here is the screenshot (ms edge view from click_in_demo.html)  which  I
used for my tests.
- I first searched the full text from the picture (sikuli area with robot
framework) : robot framework returns an empty screen without my changes
-  searched for "Right" and "Left" text : robot framework indicates "Right"
and "Left" not found on the screen without my changes
- I added html buttons with text in french, i added a sikuli robot
framework keyword changing the ocr language in french (my native language).
After changes, sikuli returns "Gauche" and "Hier" locations.

Best regards,

Pierre

Le ven. 13 nov. 2020 à 10:03, Raimund Hocke <notifications@github.com> a
écrit :

> @pierre6677 <https://github.com/pierre6677>
> I made some tests with 2.0.4 and the dev version 2.0.5: I had no problems.
>
> So again: please provide sample screenshots and sample images/texts.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RaiMan/SikulixAPI/pull/1#issuecomment-726640081>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG77AD7NKPOJV4VB6WSM4QLSPTY7DANCNFSM4TSHHB4Q>
> .
>

RaiMan commented:
Sorry, but I cannot see an image.

Please edit you answer directly on GitHub.

pierre6677 commented:
![click_in](https://user-images.githubusercontent.com/29356047/99150773-35d4fc80-2697-11eb-84c9-93ee5533a7f2.png)

RaiMan commented:
ok, thanks.

RaiMan commented:
Thanks for the sample. I also found the HTML, that you used to create this button setup on the SikuliLib site - so sorry fo the buzz.

I made some tests, and can confirm, that it does not work, if the region to look in contains all button lines or one button line completely.

If I restrict the region to only parts of it (max 3 lines or only 2 buttons, it works most of the times.

So I agree, that the button edges somehow lead to the problem, that OCR does not recognise lines of text in this special situation.

As mentioned earlier, if the space between the text and the top/bottom button edges is larger, it works.

I accept this as bug/enhancement request and will follow up here:
https://github.com/RaiMan/SikuliX1/issues/388

All the best
Raimund aka RaiMan

Am 14.11.2020 um 16:34 schrieb pierre6677 <notifications@github.com<mailto:notifications@github.com>>:



[click_in]<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F29356047%2F99150773-35d4fc80-2697-11eb-84c9-93ee5533a7f2.png&data=04%7C01%7C%7Cf675effb30514d00635708d888b2c4ae%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637409648663939246%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=axWSjJ%2B84%2B%2Br1l7hNGcWsSd4besDcjZVV9sSXAz27gM%3D&reserved=0>

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FRaiMan%2FSikulixAPI%2Fpull%2F1%23issuecomment-727224055&data=04%7C01%7C%7Cf675effb30514d00635708d888b2c4ae%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637409648663949240%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=gbeEUvIok9Q%2FVPEZ6I8NjSDdqWoMNOJ07rN9LaM0I4w%3D&reserved=0>, or unsubscribe<https://eur05.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAEPFAQQCXKL2FFM7SX26ADSP2PQBANCNFSM4TSHHB4Q&data=04%7C01%7C%7Cf675effb30514d00635708d888b2c4ae%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637409648663949240%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=PxAjddTFHi5AmZGTDsQ6zz%2BAXxxmSnwBAcySUqgt%2F1I%3D&reserved=0>.
